File: catalogue-application/src/main/java/io/quartic/catalogue/inmemory/InMemoryBackendConfig.java
package io.quartic.catalogue.inmemory;

import io.quartic.catalogue.StorageBackend;
import io.quartic.catalogue.StorageBackendConfig;

public class InMemoryBackendConfig implements StorageBackendConfig {
    @Override
    public StorageBackend build() {
        return new InMemoryStorageBackend();
    }
}

File: catalogue/src/main/kotlin/io/quartic/catalogue/inmemory/InMemoryBackendConfig.kt
package io.quartic.catalogue.inmemory

import io.dropwizard.setup.Environment
import io.quartic.catalogue.StorageBackendConfig
import io.quartic.common.secrets.SecretsCodec

class InMemoryBackendConfig : StorageBackendConfig {
    override fun build(env: Environment, secretsCodec: SecretsCodec) = InMemoryStorageBackend()
}
